### DISCUSSION TITLE
Search improvement: narrow to the scope of the current docset

### DISCUSSION THREAD

**Discussion by suetanvil** [zealdocs/zeal#1526] [Ideas]
Hello,

I'd like to request a new feature. I would really like the search to be (optionally?) limited to the currently selected section of the current docset.

Yesterday, I saw the word `use` in some unfamiliar Ruby code.  I wondered if this was part of the Ruby standard library, so I did a Zeal search.  Since I have about ten docsets installed, the results of this were unusable because **every** identifier in **every** docset containing the character sequence 'u', 's', and 'e' in showed up; furthermore, they were not ordered usefully (e.g. grouped by docset).  This turned out to be useless to me and I had to resort to using the online Ruby docs.  (Answer: no, `use` is not part of the library.)

I would like some way to restrict the scope of the search the relevant section.  The most intuitive way (I think) would be to restrict it to whichever section of the docset is currently selected on the left pane, so that's my idea.

Apologies if this is  something Zeal already does and I just missed the relevant setting.

**Comment by livingsilver94** [comment]
Zeal already does that. In the search bar, write "ru" and then hit TAB. Zeal should autocomplete it with "ruby:". That's how you'd restrict a search in a docset: by prefixing your keyword with the docset name and a colon.

Also see the readme: https://github.com/zealdocs/zeal#query--filter-docsets

**Reply by suetanvil** [reply]
Ah, thanks.

Note that I installed Zeal via the Ubuntu package so this information wasn't immediately available.  It certainly wasn't mentioned in any of Zeal's online help, as far as I could see. It might be worthwhile to add something like this if enough people run into this problem.

**Comment by Antix-Development** [comment]
I'll squeak up and say that I had to google to discover that I had to type "javascript:" to find things pertaining to javascript. It would be nice if just typing "javascript" would accomplish the same :)

Having said that, the information was on the front page :D

**Comment by trollixx** [comment]
As others pointed out this is already supported. In 07a357cab00ae6b9cd999eed27ce00be938ee9fb search prefixes will be displayed in the Docset Manager window. This will be part of the next release.

![Screenshot](https://github.com/user-attachments/assets/0e032212-a175-48fc-a5a1-d14a15bb6c9c)
